### PR TITLE
update the docs of new feature in cdc v4.0.5

### PR DESCRIPTION
**Pull request by leoppro**
### What is changed, added or deleted? (Required)

update the docs of new feature in cdc v4.0.5

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: ticdc/manage-ticdc.md [hunk 1]
```diff
@@ -431,6 +431,9 @@ curl -X POST curl 127.0.0.1:8300/capture/owner/move_table -X POST -d 'cf-id=cf06
 # 该配置会同时影响 filter 和 sink 相关配置，默认为 true
 case-sensitive = true
 
+# 是否输出 old value
```
Comment (leoppro) [inline ticdc/manage-ticdc.md:434]:
```suggestion
# 是否输出 old value，从 v4.0.5 开始支持
```

File: ticdc/ticdc-open-protocol.md [hunk 2]
```diff
@@ -68,30 +68,91 @@ Value:
...
+
+| 参数         | 类型   | 说明                    |
+| :---------- | :----- | :--------------------- |
+| Column Name    | String | 列名   |
+| Column Type    | Number | 列类型，详见：[Column 的类型码](#column-的类型码) |
+| ~~Where Handle~~   | Bool   | ~~表示该列是否可以作为 Where 筛选条件，当该列在表内具有唯一性时，Where Handle 为 true。~~ 已弃用，使用 Flag 中的 HandleKeyFlag 代替 |
+| Flag           | Number | 列标志位，详见：[列标志位](#列标志位) |
+| Column Value   | Any    | 列值   |
+
+> 注：
```
Comment (leoppro) [inline ticdc/ticdc-open-protocol.md:154]:
```suggestion
```

File: ticdc/ticdc-open-protocol.md [hunk 3]
```diff
@@ -68,30 +68,91 @@ Value:
...
+| 参数         | 类型   | 说明                    |
+| :---------- | :----- | :--------------------- |
+| Column Name    | String | 列名   |
+| Column Type    | Number | 列类型，详见：[Column 的类型码](#column-的类型码) |
+| ~~Where Handle~~   | Bool   | ~~表示该列是否可以作为 Where 筛选条件，当该列在表内具有唯一性时，Where Handle 为 true。~~ 已弃用，使用 Flag 中的 HandleKeyFlag 代替 |
+| Flag           | Number | 列标志位，详见：[列标志位](#列标志位) |
+| Column Value   | Any    | 列值   |
+
+> 注：  
+> 1. 在 Update 事件中，只有开启 OldValue 特性，才会
```
Comment (leoppro) [inline ticdc/ticdc-open-protocol.md:155]:
```suggestion
```

File: ticdc/ticdc-open-protocol.md [hunk 4]
```diff
@@ -289,3 +352,38 @@ Column 和 DDL 的类型码是由 TiCDC Open Protocol 定义的 Column 和 DDL 
 | Create Sequence                   | 34 |
 | Alter Sequence                    | 35 |
 | Drop Sequence                     | 36 |
+
```
Comment (leoppro) [inline ticdc/ticdc-open-protocol.md:355]:
```suggestion
```

File: ticdc/ticdc-open-protocol.md [hunk 5]
```diff
@@ -289,3 +350,37 @@ Column 和 DDL 的类型码是由 TiCDC Open Protocol 定义的 Column 和 DDL 
...
+若某列 Flag 值为 46，则代表这一列为组合索引列、主键列、生成列、Handle 列。
+
+```
+46 == 0b_010_1110
+   == MultipleKeyFlag | PrimaryKeyFlag | GeneratedColumnFlag | HandleKeyFlag
+```
+
+> 注:  
+> 1. BinaryFlag 仅在列为 Blob/Text（包括 Tiny Blob/Tiny Blob、Long Blob/Long Blob 等） 类型时才有意义。当上游列为 Text 时，BinaryFlag 置 1，当上游列为 Blob 时，BinaryFlag 置 0。
```
Comment (amyangfei) [inline ticdc/ticdc-open-protocol.md:385]: 这个没反么，Text 时 BinaryFlag -> 0 ？

Comment (leoppro) [reply ticdc/ticdc-open-protocol.md:385]: 哦 反了。。。

Comment (leoppro) [inline ticdc/ticdc-open-protocol.md:385]:
```suggestion
> 1. BinaryFlag 仅在列为 Blob/Text（包括 Tiny Blob/Tiny Blob、Long Blob/Long Blob 等） 类型时才有意义。当上游列为 Blob 时，BinaryFlag 置 1，当上游列为 Text 时，BinaryFlag 置 0。
```

File: ticdc/ticdc-overview.md [hunk 6]
```diff
@@ -66,10 +66,12 @@ TiCDC 的系统架构如下图所示：
...
 ## 同步限制
 
-将数据同步到 TiDB 或 MySQL，需要满足以下条件才能保证正确性：
+TiCDC 只能同步至少存在一个**有效索引**的表，**有效索引**的定义如下：
 
-- 表必须要有主键或者唯一索引。
-- 如果表只存在唯一索引，至少有一个唯一索引的每一列在表结构中明确定义 `NOT NULL`。
+- 主键（`PRIMARY KEY`）为有效索引。
+- 同时满足下列条件的唯一索引（`UNIQUE INDEX`）为有效索引：
+  - 索引中每一列在表结构中定义明确定义非空（ `NOT NULL`）。
```
Comment (leoppro) [inline ticdc/ticdc-overview.md:73]:
```suggestion
  - 索引中每一列在表结构中定义明确定义非空（`NOT NULL`）。
```

File: ticdc/ticdc-open-protocol.md [hunk 7]
```diff
@@ -68,30 +68,89 @@ Value:
 
 + **Value:**
 
+
+    Insert 事件，输出新增的行数据。
```
Comment (TomShawn) [inline ticdc/ticdc-open-protocol.md:72]:
```suggestion
```

**Comment by TomShawn** [comment]
Please also convert the .svg format to .png.

File: ticdc/ticdc-open-protocol.md [hunk 8]
```diff
@@ -289,3 +350,37 @@ Column 和 DDL 的类型码是由 TiCDC Open Protocol 定义的 Column 和 DDL 
...
+| 2   | 0x02 | HandleKeyFlag       | 该列是否为 Handle 列 |
+| 3   | 0x04 | GeneratedColumnFlag | 该列是否为生成列      |
+| 4   | 0x08 | PrimaryKeyFlag      | 该列是否为主键列      |
+| 5   | 0x10 | UniqueKeyFlag       | 该列是否为唯一索引列   |
+| 6   | 0x20 | MultipleKeyFlag     | 该列是否为组合索引列   |
+| 7   | 0x40 | NullableFlag        | 该列是否为可空列          |
+
+示例：
+
+若某列 Flag 值为 85，则代表这一列为可空列、唯一索引列、生成列、二进制编码列。
```
[the remark this answers is not in the document]
Comment (leoppro) [reply ticdc/ticdc-open-protocol.md:369]: 是且的关系

File: ticdc/ticdc-open-protocol.md [hunk 9]
```diff
@@ -68,30 +68,88 @@ Value:
 
 + **Value:**
 
+    `Insert` 事件，输出新增的行数据。
```
Comment (TomShawn) [inline ticdc/ticdc-open-protocol.md:71]:
```suggestion
    `Insert` 事件，输出新增的行数据。

```

File: ticdc/ticdc-open-protocol.md [hunk 10]
```diff
@@ -68,30 +68,88 @@ Value:
...
     ```
 
-    | 参数         | 类型   | 说明                    |
-    | :---------- | :----- | :--------------------- |
-    | UpdateOrDelete | String | 标识该 Event 是增加 Row 还是删除 Row，取值只可能是 "u"/"d" |
-    | Column Name    | String | 列名   |
-    | Column Type    | Number | 列类型，详见：[Column 和 DDL 的类型码](#column-和-ddl-的类型码) |
-    | Where Handle   | Bool   | 表示该列是否可以作为 Where 筛选条件，当该列在表内具有唯一性时，Where Handle 为 true |
-    | Column Value   | Any    | 列值   |
+    `Update` 事件，输出新增的行数据 ("u") 以及修改前的行数据 ("p")，仅当 Old Value 特性开启时，才会输出修改前的行数据。
```
Comment (TomShawn) [inline ticdc/ticdc-open-protocol.md:91]:
```suggestion
    `Update` 事件，输出新增的行数据 ("u") 以及修改前的行数据 ("p")，仅当 Old Value 特性开启时，才会输出修改前的行数据。

```

File: ticdc/ticdc-open-protocol.md [hunk 11]
```diff
@@ -68,30 +68,88 @@ Value:
...
+                "t":<Column Type>,
+                "h":<Where Handle>,
+                "f":<Flag>,
+                "v":<Column Value>
+            }
+        }
+    }
+    ```
+
+    `Delete` 事件，输出被删除的行数据。当 Old Value 特性开启时，`Delete` 事件中包含被删除的行数据中的所有列；当 Old Value 特性关闭时，`Delete` 事件中仅包含 [HandleKey](#列标志位) 列。
```
Comment (TomShawn) [inline ticdc/ticdc-open-protocol.md:125]:
```suggestion
    `Delete` 事件，输出被删除的行数据。当 Old Value 特性开启时，`Delete` 事件中包含被删除的行数据中的所有列；当 Old Value 特性关闭时，`Delete` 事件中仅包含 [HandleKey](#列标志位) 列。

```

File: ticdc/ticdc-open-protocol.md [hunk 12]
```diff
@@ -68,30 +68,88 @@ Value:
...
+    ```
+
+| 参数         | 类型   | 说明                    |
+| :---------- | :----- | :--------------------- |
+| Column Name    | String | 列名   |
+| Column Type    | Number | 列类型，详见：[Column 的类型码](#column-的类型码) |
+| Where Handle（**已弃用**）   | Bool   | 表示该列是否可以作为 Where 筛选条件，当该列在表内具有唯一性时，Where Handle 为 true。该参数已弃用，被 Flag 中的 HandleKeyFlag 代替 |
+| Flag           | Number | 列标志位，详见：[列标志位](#列标志位) |
+| Column Value   | Any    | 列值   |
+
```
Comment (TomShawn) [inline ticdc/ticdc-open-protocol.md:152]:
```suggestion
```

File: ticdc/ticdc-overview.md [hunk 13]
```diff
@@ -66,10 +66,12 @@ TiCDC 的系统架构如下图所示：
...
 ## 同步限制
 
-将数据同步到 TiDB 或 MySQL，需要满足以下条件才能保证正确性：
+TiCDC 只能同步至少存在一个**有效索引**的表，**有效索引**的定义如下：
 
-- 表必须要有主键或者唯一索引。
-- 如果表只存在唯一索引，至少有一个唯一索引的每一列在表结构中明确定义 `NOT NULL`。
+- 主键 (`PRIMARY KEY`) 为有效索引。
+- 同时满足下列条件的唯一索引 (`UNIQUE INDEX`) 为有效索引：
+  - 索引中每一列在表结构中明确定义非空 (`NOT NULL`)。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:73]:
```suggestion
    - 索引中每一列在表结构中明确定义非空 (`NOT NULL`)。
```

File: ticdc/ticdc-overview.md [hunk 14]
```diff
@@ -66,10 +66,12 @@ TiCDC 的系统架构如下图所示：
...
+  - 索引中不存在虚拟生成列 (`VIRTUAL GENERATED COLUMNS`)。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:74]:
```suggestion
    - 索引中不存在虚拟生成列 (`VIRTUAL GENERATED COLUMNS`)。
```

**Pull request merged**